Assertion error in `toAnchor`

Compiling 
```scala
 xml"""<foo bar=${"Hello"}/>"""
```
with scalahost triggers an assertion error https://gist.github.com/allanrenucci/83db9369e47fd3eb3d1a599c598dc7d5

```
import scala.xml.quote._
object A {
  xml"""<foo bar=${"Hello"}/>"""
}
java.lang.AssertionError: assertion failed
	at scala.Predef$.assert(Predef.scala:204)
	at scala.meta.internal.scalahost.databases.AnchorOps$XtensionGPositionAnchor.toAnchor(AnchorOps.scala:25)
	at scala.meta.internal.scalahost.databases.SymbolOps$XtensionGSymbolMSymbol.toSemantic(SymbolOps.scala:26)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.success$1(AttributedSourceOps.scala:142)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.tryMstart$1(AttributedSourceOps.scala:172)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.tryFindMtree(AttributedSourceOps.scala:219)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.traverse(AttributedSourceOps.scala:316)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.traverse(AttributedSourceOps.scala:136)
	at scala.reflect.api.Trees$Traverser.$anonfun$traverseTrees$1(Trees.scala:2484)
	at scala.reflect.api.Trees$Traverser.traverseTrees(Trees.scala:2484)
	at scala.reflect.internal.Trees.traverseComponents$1(Trees.scala:1238)
	at scala.reflect.internal.Trees.itraverse(Trees.scala:1334)
	at scala.reflect.internal.Trees.itraverse$(Trees.scala:1204)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.traverse(AttributedSourceOps.scala:322)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.traverse(AttributedSourceOps.scala:136)
	at scala.reflect.internal.Trees.traverseComponents$1(Trees.scala:1281)
	at scala.reflect.internal.Trees.itraverse(Trees.scala:1334)
	at scala.reflect.internal.Trees.itraverse$(Trees.scala:1204)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.traverse(AttributedSourceOps.scala:322)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.traverse(AttributedSourceOps.scala:136)
	at scala.reflect.api.Trees$Traverser.$anonfun$traverseStats$2(Trees.scala:2498)
	at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
	at scala.reflect.api.Trees$Traverser.$anonfun$traverseStats$1(Trees.scala:2498)
	at scala.reflect.api.Trees$Traverser.traverseStats(Trees.scala:2497)
	at scala.reflect.internal.Trees.traverseComponents$1(Trees.scala:1236)
	at scala.reflect.internal.Trees.itraverse(Trees.scala:1334)
	at scala.reflect.internal.Trees.itraverse$(Trees.scala:1204)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.traverse(AttributedSourceOps.scala:322)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.traverse(AttributedSourceOps.scala:136)
	at scala.reflect.internal.Trees.$anonfun$itraverse$1(Trees.scala:1212)
	at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
	at scala.reflect.internal.Trees.traverseMemberDef$1(Trees.scala:1207)
	at scala.reflect.internal.Trees.itraverse(Trees.scala:1331)
	at scala.reflect.internal.Trees.itraverse$(Trees.scala:1204)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.traverse(AttributedSourceOps.scala:322)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.traverse(AttributedSourceOps.scala:136)
	at scala.reflect.api.Trees$Traverser.$anonfun$traverseStats$2(Trees.scala:2498)
	at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
	at scala.reflect.api.Trees$Traverser.$anonfun$traverseStats$1(Trees.scala:2498)
	at scala.reflect.api.Trees$Traverser.traverseStats(Trees.scala:2497)
	at scala.reflect.internal.Trees.itraverse(Trees.scala:1330)
	at scala.reflect.internal.Trees.itraverse$(Trees.scala:1204)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.traverse(AttributedSourceOps.scala:322)
	at scala.meta.internal.scalahost.databases.AttributedSourceOps$XtensionCompilationUnitAttributedSource$traverser$4$.traverse(AttributedSourceOps.scala:136)
	at scala.reflect.api.Trees$Traverser.$anonfun$traverseStats$2(Trees.scala:2498)
	at scala.reflect.api.Trees$Traverser.atOwner(Trees.scala:2507)
	at scala.reflect.api.Trees$Traverser.$anonfun$traverseStats$1(Trees.scala:2498)
	at scala.reflect.api.Trees$Traverser.traverseStats(Trees.scala:2497)
	at scala.reflect.internal.Trees.itraverse(Trees.scala:1330)
	at scala.reflect.internal.Trees.itraverse$(Trees.scala:1204)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.internal.SymbolTable.itraverse(SymbolTable.scala:16)
	at scala.reflect.api.Trees$Traverser.traverse(Trees.scala:2475)
```

Discovered at https://github.com/scalacenter/scalafix/pull/146

I am able to reproduce this error while running latest sbt-scalahost on the community build

```scala
java.lang.AssertionError: assertion failed
        at scala.Predef$.assert(Predef.scala:156)
        at scala.meta.internal.semantic.InputOps$XtensionGPositionMPosition.toMeta(InputOps.scala:20)
        at scala.meta.internal.semantic.SymbolOps$XtensionGSymbolMSymbol.toSemantic(SymbolOps.scala:25)
        at scala.meta.internal.semantic.AttributesOps$XtensionCompilationUnitAttributes$$anonfun$toAttributes$1$traverser$4$.scala$meta$internal$semantic$A
ttributesOps$XtensionCompilationUnitAttributes$$anonfun$traverser$$success$1(AttributesOps.scala:141)
        at scala.meta.internal.semantic.AttributesOps$XtensionCompilationUnitAttributes$$anonfun$toAttributes$1$traverser$4$.tryMpos$1(AttributesOps.scala:
183)
        at scala.meta.internal.semantic.AttributesOps$XtensionCompilationUnitAttributes$$anonfun$toAttributes$1$traverser$4$.tryFindMtree(AttributesOps.sca
la:203)
        at scala.meta.internal.semantic.AttributesOps$XtensionCompilationUnitAttributes$$anonfun$toAttributes$1$traverser$4$.traverse(AttributesOps.scala:3
22)
```

--- a/scalahost/nsc/src/main/scala/scala/meta/internal/semantic/InputOps.scala
+++ b/scalahost/nsc/src/main/scala/scala/meta/internal/semantic/InputOps.scala
@@ -17,9 +17,15 @@ trait InputOps { self: DatabaseOps =>
 
   implicit class XtensionGPositionMPosition(pos: GPosition) {
     def toMeta: m.Position = {
-      assert(pos.isRange)
       val input = m.Input.File(pos.source.toAbsolutePath)
-      m.Position.Range(input, m.Point.Offset(input, pos.start), m.Point.Offset(input, pos.end))
+      if (pos.isRange) {
+        m.Position.Range(input, m.Point.Offset(input, pos.start), m.Point.Offset(input, pos.end))
+      } else {
+        // NOTE: Even with -Yrangepos enabled we cannot be guaranteed that all positions are
+        // range positions. In the case we encounter a non-range position we assume start == end.
+        val mpoint = m.Point.Offset(input, pos.point)
+        m.Position.Range(input, mpoint, mpoint)
+      }
     }
   }
 }

